### DISCUSSION TITLE
am i using it efficiently?

### DISCUSSION THREAD

**Discussion by azinsharaf** [CRAG666/code_runner.nvim#82] [Q&A] [answered]
hi, i wanna make sure i am using this great plugin efficiently. I have a python code and to run it i press `<leader>rr`. It runs the code. Then i edit the code and want to run it again. I have to close the runner first ( `<leader>rc`) and then run it `<leader>rr`. It means I press 6 keys for it. Am i doing it right? Why can't i re-run the code with 3 keys? Maybe i need to merge the commands together? 

```
return {
  "CRAG666/code_runner.nvim",
  event = "VeryLazy",
  init = function() require("which-key").register({ ["<leader>r"] = { name = "Code Runner" } }, { mode = "n" }) end,
  opts = {
    mode = "toggle",
    focus = true,
    term = {
      position = "bot",
      size = 25,
    },
    filetype = {
      python = "C:/Users/azin/AppData/Local/ESRI/conda/envs/arcgispro-py3-clone/python.exe"
    },
  },
  keys = {
    { "<leader>rr", "<cmd>RunCode<cr>",    desc = "Run Code" },
    { "<leader>rf", "<cmd>RunFile<cr>",    desc = "Run File" },
    -- { "<leader>rft", "<cmd>RunFile tab<cr>", desc = "Run File Tab" },
    { "<leader>rp", "<cmd>RunProject<cr>", desc = "Run Project" },
    { "<leader>rc", "<cmd>RunClose<cr>",   desc = "Run Close" },
    -- { "<leader>crf", "<cmd>CRFiletype<cr>", desc = "Open Json supported files" },
    -- { "<leader>crp", "<cmd>CRProject<cr>", desc = "Open Json list of projects" },
  },
}
```

**Comment by CRAG666** [comment]
You don't need it, just press \<leader\>rr again

**Reply by azinsharaf** [reply]
@CRAG666 But when I press `<leader>rr` again it closes the code_runner windows instead of re-running it. Is it different what it does on your system?

https://github.com/CRAG666/code_runner.nvim/assets/10101468/0246b790-f9d6-4f61-aeaa-7dc3e872e871

**Reply by CRAG666** [reply]
I think the problem is that it uses toggle mode, try using some other mode

**Comment by azinsharaf** [comment]
the only mode works well is float. Is it correct? I can close it by pressing escape and run it again.


https://github.com/CRAG666/code_runner.nvim/assets/10101468/8a9c48e2-bfd4-4b6a-8cac-fa401083eec5

**Reply by CRAG666** [reply, marked as answer]
Not just float all modes except toggle, could work to enable toggle, but I'd rather use betterterm as it already has those features
